llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -249,7 +249,8 @@ namespace {
   public:
     DAGCombiner(SelectionDAG &D, AliasAnalysis *AA, CodeGenOptLevel OL)
```

Comment:
It would be better to directly accept a BatchAA argument here. That way you can reuse the same BatchAA instance across multiple DAGCombine runs (we do 2-4 per function).